app/collectors/acmCertificates.scala

```diff
@@ -0,0 +1,114 @@
+    .withCredentials(origin.credentials.provider)
+    .withRegion(origin.awsRegion)
+    .build()
+
+  private def crawlWithMarker(token: Option[String]): Iterable[AcmCertificate] = {
+    val request = new ListCertificatesRequest().withNextToken(token.orNull)
+    val result = client.listCertificates(request)
+    val configs = result.getCertificateSummaryList.asScala.map { cert =>
+      val requestDetails = new DescribeCertificateRequest().withCertificateArn(cert.getCertificateArn)
+      val resultDetails = client.describeCertificate(requestDetails)
+      AcmCertificate.fromApiData(resultDetails.getCertificate, origin)
+    }
+    Option(result.getNextToken) match {
+      case None => configs
+      case t@Some(_) => configs ++ crawlWithMarker(t)
+    }
+  }
+
+  def crawl: Iterable[AcmCertificate] = crawlWithMarker(None)
+}
+
+case class DomainValidation(
+                             domainName: String,
+                             validationEmails: List[String],
+                             validationDomain: String,
+                             validationStatus: String
+                           )
+
+object DomainValidation {
+  def fromApiData(dv: AwsDomainValidation): DomainValidation = {
+    DomainValidation(
+      dv.getDomainName,
+      dv.getValidationEmails.asScala.toList,
+      dv.getValidationDomain,
+      dv.getValidationStatus
+    )
+  }
+}
+
+case class RenewalInfo(renewalStatus: String, domainValidationOptions: List[DomainValidation])
+
+object RenewalInfo {
+  def fromApiData(ri: RenewalSummary): RenewalInfo = {
+    RenewalInfo(ri.getRenewalStatus, ri.getDomainValidationOptions.asScala.map(DomainValidation.fromApiData).toList)
+  }
+}
+
+object AcmCertificate {
+  def fromApiData(cert: CertificateDetail, origin: AmazonOrigin) = AcmCertificate(
+    arn = cert.getCertificateArn,
+    domainName = cert.getDomainName,
+    subjectAlternativeNames = cert.getSubjectAlternativeNames.asScala.toList,
+    certificateType = cert.getType,
+    status = cert.getStatus,
+    issuer = cert.getIssuer,
+    inUseBy = cert.getInUseBy.asScala.toList,
+    notBefore = Option(cert.getNotBefore).flatMap(dt => Try(new DateTime(dt)).toOption),
+    notAfter = Option(cert.getNotAfter).flatMap(dt => Try(new DateTime(dt)).toOption),
+    createdAt = Option(cert.getCreatedAt).flatMap(dt => Try(new DateTime(dt)).toOption),
+    issuedAt = Option(cert.getIssuedAt).flatMap(dt => Try(new DateTime(dt)).toOption),
+    failureReason = Option(cert.getFailureReason),
+    subject = cert.getSubject,
+    keyAlgorithm = cert.getKeyAlgorithm,
+    signatureAlgorithm = cert.getSignatureAlgorithm,
+    serial = cert.getSerial,
+    domainValidationOptions = cert.getDomainValidationOptions.asScala.toList.map(DomainValidation.fromApiData),
+    renewalStatus = Option(cert.getRenewalSummary).map(RenewalInfo.fromApiData)
+  )
+}
+
+case class AcmCertificate(
+                              arn: String,
```

Comment:
I think you need to fix your IntelliJ settings so it indents case class arglists properly. This is an example of where IntelliJ's default is incorrect.

Not super important here, just an FYI.

<img width="765" alt="screen shot 2014-12-23 at 15 48 58" src="https://user-images.githubusercontent.com/29761/28129739-671fe544-672c-11e7-9d13-2d6816515fc8.png">